src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -96,6 +96,55 @@ public void addEdge(ExpressionEdge e) {
   public ExpressionGraph() {
   }
 
+  /**
+   * Adds the provided subgraph to this graph
+   * Precondition: there are no variable id conflicts between subGraph and this graph. That is
+   *               each variable identifier in the subgraph is generated WRT its scope.
+   *
+   * In the final graph, subGraphInput/subGraphOutput will no longer exist, and the edges from/to them
+   * should be from mainGraphInput/mainGraphOutput respectively.
+   *
+   * The edge from mainGraphInput to the function vertex & the edge from the function vertex to mainGraphOutput
+   * are not removed (since 1 vertex can potentially point to multiple functions), so they should be deleted by
```

Comment:
This was what I was kinda getting at while outside the SE office. Suppose we had vertices:

S1 (reference), T1 (reference), F1, F2 (functions). Is it possible to have a graph like this:

S1 -> F1
S1 -> F2
F1 -> T1
F2 -> T1

In this case, since there are multiple edges from S1 to function calls, this function would not know which vertex the subgraph represents.